### PR TITLE
Feature: ignore spaces in annotation target

### PR DESCRIPTION
**Pull request by ralfhandl**

### CODE REVIEW

**Review by HeikoTheissen — DISMISSED** [review]

**Comment by HeikoTheissen** [comment]
The `edmx.xsd` does not accept whitespace around the comma:
> cvc-pattern-valid: Value `One.OddWaldos(Collection(One.Waldo), One.Waldo)` is not facet-valid with respect to pattern `[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}(([.,#(]|/@?|\(?\){1,}((,|/@?))?)[\p{L}\p{Nl}_][\p{L}\p{Nl}\p{Nd}\p{Mn}\p{Mc}\p{Pc}\p{Cf}]{0,}){0,}\(?\){0,}(/$ReturnType)?` for type `TTarget`.

Must we fix that as well?

**Comment by ralfhandl** [comment]
> Must we fix that as well?

No, whitespace is not allowed in annotation targets, this is a bug in the MS Graph EDMX.

I just want to make the EDMX converter more tolerant, and not relax the specification.

File: lib/xml2json.js [hunk 1]
```diff
@@ -1477,7 +1477,7 @@ module.exports.xml2json = function (
    */
   function normalizeTarget(target) {
     const open = target.indexOf("(");
-    const close = target.indexOf(")");
+    const close = target.lastIndexOf(")");
```
Comment (ralfhandl) [inline lib/xml2json.js:1480]: Why is that better? We don't have nested parentheses in annotation targets yet, and if we get them, this code will have to be reworked anyway.

If you found a situation where the old code breaks and the new code works better, please add a test case.

File: test/xml2json.test.js [hunk 2]
```diff
@@ -358,6 +358,9 @@ describe("Edge cases", function () {
           <Annotations Target="microsoft.graph.uploadDepToken(microsoft.graph.depOnboardingSetting, Edm.String, Edm.String)">
             <Annotation Term="Org.OData.Core.V1.Description" String="Uploads a new Device Enrollment Program token" />
           </Annotations>
+          <Annotations Target="One.OddWaldos(Collection(One.Waldo), One.Waldo)">
```
Comment (ralfhandl) [inline test/xml2json.test.js]: Thanks, good catch!

**Review by HeikoTheissen — APPROVED** [review]

**Review by mikepizzo — APPROVED** [review]
